### PR TITLE
RELATED: BB-849 Support for the previous period measure definition

### PR DESCRIPTION
**Pull request by xMort**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/27/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/20/ : FAILURE in 1m 29s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/28/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/21/ : SUCCESS in 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/29/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/22/ : SUCCESS in 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/35/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/30/ : SUCCESS in 1m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/42/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/37/ : SUCCESS in 2m 08s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/44/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/39/ : SUCCESS in 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/45/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/40/ : SUCCESS in 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/46/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/41/ : SUCCESS in 1m 40s

File: src/DataLayer/utils/AfmUtils.ts [hunk 1]
```diff
@@ -50,6 +61,17 @@ export function normalizeAfm(afm: AFM.IAfm): INormalizedAFM {
...
 }
 
+/**
+ * Returns true if measure is a simple measure
+ *
+ * @method isSimpleMeasure
+ * @param {AFM.IMeasure} item
+ * @returns {boolean}
+ */
+export const isSimpleMeasure = (item: AFM.IMeasure): boolean => {
```
Comment (martinmosko) [inline src/DataLayer/utils/AfmUtils.ts]: use function instead of arrow

File: src/DataLayer/utils/AfmUtils.ts [hunk 2]
```diff
@@ -61,6 +83,17 @@ export const isPoP = (item: AFM.IMeasure): boolean => {
...
 };
 
+/**
+ * Returns true if measure is previous period measure
+ *
+ * @method isPreviousPeriodMeasure
+ * @param {AFM.IMeasure} item
+ * @returns {boolean}
+ */
+export const isPreviousPeriodMeasure = (item: AFM.IMeasure): boolean => {
```
Comment (martinmosko) [inline src/DataLayer/utils/AfmUtils.ts]: use function instead of arrow

File: src/DataLayer/utils/AfmUtils.ts (hunk 1 above)
Comment (xMort) [reply src/DataLayer/utils/AfmUtils.ts]: Fixed

File: src/DataLayer/utils/AfmUtils.ts (hunk 2 above)
Comment (xMort) [reply src/DataLayer/utils/AfmUtils.ts]: Fixed. For the `isPoPMeasure` as well.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/47/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/42/ : SUCCESS in 55s

File: src/DataLayer/converters/MeasureConverter.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    return {
+        localIdentifier: measure.measure.localIdentifier,
+        definition: convertedDefinition,
+        ...aliasProp,
+        ...formatProp
+    };
+}
+
+function convertMeasureDefinition(definition: IMeasureDefinitionType): AFM.MeasureDefinition {
+    if (VisualizationObject.isMeasureDefinition(definition)) {
```
Comment (martinmosko) [inline src/DataLayer/converters/MeasureConverter.ts]: `isSimpleMeasureDefinition` ?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/48/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/43/ : SUCCESS in 1m 09s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/11/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/44/ : SUCCESS in 45s

**Pull request merged**